src/v/iceberg/manifest_entry_values.cc

```diff
@@ -0,0 +1,249 @@
+    auto ret = std::get<std::unique_ptr<struct_value>>(std::move(*v));
+    if (!ret) {
+        throw std::invalid_argument("Struct value is nullptr");
+    }
+    return ret;
+}
+
+chunked_hash_map<nested_field::id_t, size_t>
+get_counts_map(std::optional<value> v) {
+    if (!v.has_value()) {
+        return {};
+    }
+    if (!holds_alternative<std::unique_ptr<map_value>>(*v)) {
+        throw std::invalid_argument("Value is not a map");
+    }
+    auto& as_map = std::get<std::unique_ptr<map_value>>(*v);
+    chunked_hash_map<nested_field::id_t, size_t> ret;
+    for (auto& kv : as_map->kvs) {
+        auto k = get_required_primitive<int_value>(std::move(kv.key));
+        auto v = get_required_primitive<long_value>(std::move(kv.val));
+        ret.emplace(nested_field::id_t{k}, v);
+    }
+    return ret;
+}
+
+template<typename ValueT, typename T>
+std::optional<value> to_optional_value(std::optional<T> v) {
+    if (!v.has_value()) {
+        return std::nullopt;
+    }
+    return ValueT{*v};
+}
+
+int status_to_int(manifest_entry_status s) {
+    switch (s) {
+    case manifest_entry_status::existing:
+        return 0;
+    case manifest_entry_status::added:
+        return 1;
+    case manifest_entry_status::deleted:
+        return 2;
+    }
+}
+
+manifest_entry_status status_from_int(int s) {
+    if (s == 0) {
+        return manifest_entry_status::existing;
+    }
+    if (s == 1) {
+        return manifest_entry_status::added;
+    }
+    if (s == 2) {
+        return manifest_entry_status::deleted;
+    }
+    throw std::invalid_argument(
+      fmt::format("Invalid manifest entry status: {}", s));
+}
+
+int content_to_int(data_file_content_type c) {
+    switch (c) {
+    case data_file_content_type::data:
+        return 0;
+    case data_file_content_type::position_deletes:
+        return 1;
+    case data_file_content_type::equality_deletes:
+        return 2;
+    }
+}
+
+data_file_content_type content_from_int(int c) {
+    if (c == 0) {
+        return data_file_content_type::data;
+    }
+    if (c == 1) {
+        return data_file_content_type::position_deletes;
+    }
+    if (c == 2) {
+        return data_file_content_type::equality_deletes;
+    }
+    throw std::invalid_argument(
+      fmt::format("Invalid data file content type: {}", c));
+}
+
+iobuf format_to_str(data_file_format f) {
+    switch (f) {
+    case data_file_format::avro:
+        return iobuf::from("avro");
+    case data_file_format::orc:
+        return iobuf::from("orc");
+    case data_file_format::parquet:
+        return iobuf::from("parquet");
+    }
+}
+
+data_file_format format_from_str(std::string_view s) {
+    return string_switch<data_file_format>(s)
+      .match("avro", data_file_format::avro)
+      .match("AVRO", data_file_format::avro)
+      .match("orc", data_file_format::orc)
+      .match("ORC", data_file_format::orc)
+      .match("parquet", data_file_format::parquet)
+      .match("PARQUET", data_file_format::parquet);
```

Comment:
It looks like there's a `match_all` method that could be used for this:

`.match_all("parquet", "PARQUET", data_file_format::parquet);`

It looks like the spec only mentions all lowercase. Do we want to support other cases like "Parquet" or "PaRqUeT"?

Reply:
Sure, will do in a follow up.